### DISCUSSION TITLE
Does makes sense to use like that?

### DISCUSSION THREAD

**Discussion by ImamJanjua** [pmndrs/zustand#2535] [General]
Hi, I am trying out zustand but not really sure if I am using it correctly?

<img width="696" alt="Bildschirmfoto 2024-05-06 um 20 46 01" src="https://github.com/pmndrs/zustand/assets/137432044/94b39c90-e9b0-42b2-a426-bebac0bb59e1">
<img width="850" alt="Bildschirmfoto 2024-05-06 um 20 46 51" src="https://github.com/pmndrs/zustand/assets/137432044/2f6c904f-b7dc-4c24-acb8-72e7e64505bb">

**Comment by dai-shi** [comment]
Your selector returns a new reference every time, which lose the render optimization. There are many ways to fix it, but one is to use `useShallow`.

**Comment by ImamJanjua** [comment]
@dai-shi  u mean something like this? So calling with the use syntax.

`const setDoctor = useDoctorStore.use.setDoctor();
  const email = useDoctorStore.use.email;
  const password = useDoctorStore.use.password;`

**Reply by dai-shi** [reply]
that should work too.

**Reply by ImamJanjua** [reply]
thanks, but  how can desctruture it?

**Reply by dai-shi** [reply]
They are already isolated. I'm not sure why you need to destructure them?

**Reply by ImamJanjua** [reply]
because that would a bit less lines of code especially if store has more properties. But if this the way than i will use it.

**Reply by dai-shi** [reply]
useShallow (or createWithEqualityFn + shallow) may help.

```js
const [a, b] = useFooStore(useShallow((state) => [state.a, state.b]))
```

**Comment by ImamJanjua** [comment]
Hi, me again. Could u maybe tell me why i am getting this errors when i wanna use persist?

`const useDoctorStore = create<State & Actions>(
  persist(
    (set) => ({
      ...initialState,
      setDoctor: (doctor) => {
        set((state) => ({ ...state, ...doctor }));
      },
    }),
    {
      name: "doctorStore",
      getStorage: () => sessionStorage,
    },
  ),
);`

<img width="893" alt="Bildschirmfoto 2024-05-07 um 11 38 42" src="https://github.com/pmndrs/zustand/assets/137432044/a701ec9f-733c-499d-baaa-7a3bcaae1088">

**Reply by dbritto-dev** [reply]
@ImamJanjua you should use the curried version in TypeScript `create<T>()(...)`
